### PR TITLE
Vauelidate Forgot Password page and path to it

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: client/src/components/auth/ForgotPassword.vue [hunk 1]
```diff
@@ -0,0 +1,116 @@
+<template>
+    <div>
+        <h4 class="title-of-card pb-4">Reset your password</h4>
```
Comment (lenchvolodymyr) [inline client/src/components/auth/ForgotPassword.vue:3]: localization

File: client/src/components/auth/ForgotPassword.vue [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        <p class="info-text-p">{{ lang.ENTER_YOUR_EMAIL_ADDRESS }}</p>
+        <VForm>
+            <VCol cols="11" sm="11" md="8" class="pa-0 py-4">
+                <label for="email">{{ lang.EMAIL }}*</label>
+                <VTextField
+                    id="email"
+                    v-bind:value="email"
```
Comment (lenchvolodymyr) [inline client/src/components/auth/ForgotPassword.vue:10]: you can use a shortcut: :value="email"

File: client/src/components/auth/ForgotPassword.vue [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                    :error-messages="emailErrors"
+                    @blur="setEmail($event.target.value)"
```
Comment (lenchvolodymyr) [inline client/src/components/auth/ForgotPassword.vue:12]: isn't better to use @input event? In this case the email will change each time you press key:

@input="setEmail"
```
setEmail(e) {
    this.email = e.target.value;
}
```

File: client/src/components/auth/ForgotPassword.vue [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                    <VBtn
+                        height="44"
+                        class="login-button  primary"
+                        depressed
+                        @click="onSignIn"
+                        >{{ lang.SEND_RESET_INSTRUCTION }}
+                    </VBtn>
+                    <RouterLink
+                        :to="{ name: 'SignIn' }"
+                        class="remembered-password-a "
```
Comment (lenchvolodymyr) [inline client/src/components/auth/ForgotPassword.vue:29]: what is "-a" ?

File: client/src/components/auth/ForgotPassword.vue [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            this.$v.email.$touch();
+        }
+    },
+    computed: {
+        emailErrors() {
+            const errors = [];
+            if (!this.$v.email.$dirty) {
+                return errors;
+            }
+            !this.$v.email.email && errors.push('Must be valid e-mail');
```
Comment (lenchvolodymyr) [inline client/src/components/auth/ForgotPassword.vue:67]: validation messages also must be in the localization

**Review by lenchvolodymyr — CHANGES_REQUESTED** [review]

File: client/src/components/auth/ForgotPassword.vue [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        }),
+        ...mapActions('notification', {
+            setErrorNotification: notificationActions.SET_ERROR_NOTIFICATION
+        }),
+        setEmail(e) {
+            this.email = e.target.value;
+            this.$v.email.$touch();
+        },
+        setEmailOnInput(e) {
+            this.email = e;
```
Comment (lenchvolodymyr) [inline client/src/components/auth/ForgotPassword.vue:89]: probably e.target.value?

File: client/src/components/auth/ForgotPassword.vue [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+            this.$v.email.$touch();
+        },
+        async onSubmit() {
+            this.$v.$touch();
+            if (!this.$v.$invalid) {
+                try {
+                    const dataForgot = { email: this.email };
+                    const answer = await this.forgotPassword(dataForgot);
```
Comment (lenchvolodymyr) [inline client/src/components/auth/ForgotPassword.vue:97]: move all this logic to vuex store. Here code should look like this:

```
onSubmit() {
   try {
      this.$v.$touch();

      if (this.$v.$invalid) {
          throw new Error(this.lang.PLEASE_ENTER_CORRECT_DATA);
      }
       await this.forgotPassword(dataForgot);
   } catch (error) {
       this.setErrorNotification(error);
  }
}
```

Comment (lenchvolodymyr) [reply client/src/components/auth/ForgotPassword.vue:97]: and `response` is more semantically correct than `answer`

File: client/src/components/auth/ResetPassword.vue [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+            this.confirmPassword = e.target.value;
+            this.$v.confirmPassword.$touch();
+        },
+        setConfirmedPasswordOnInput(e) {
+            this.confirmPassword = e;
+            this.$v.confirmPassword.$touch();
+        },
+        async onSubmit() {
+            this.$v.$touch();
+            if (!this.$v.$invalid) {
```
Comment (lenchvolodymyr) [inline client/src/components/auth/ResetPassword.vue:125]: the same as above

Comment (lenchvolodymyr) [reply client/src/components/auth/ResetPassword.vue:125]: move this logic to vuex actions

File: client/src/components/auth/ResetPassword.vue [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+                        password: this.password,
+                        token: this.$route.query.token
+                    };
+                    const answer = await this.resetPassword(dataPasswordReset);
+                    if ('error' in answer) {
+                        this.typeResultOfResetPassword = 'error';
+                        this.resultOfResetPassword = this.lang.ERROR_IN_PASSWORD_RESET;
+                        this.explanation = this.lang.EXPLANATION_ERROR_PASSWORD_RESET;
+                        this.helperVisibility = true;
+                        setTimeout(() => {
```
Comment (lenchvolodymyr) [inline client/src/components/auth/ResetPassword.vue:138]: get rid of timeouts. Add a link to page to the notification. User should make the decision itself

**Review by lenchvolodymyr — CHANGES_REQUESTED** [review]

File: client/src/components/auth/ForgotPassword.vue (hunk 6 above)
Comment (mv-v-7) [reply client/src/components/auth/ForgotPassword.vue:89]: event Input work in Vuetify in that strange way it returns value of field not object.

File: client/src/components/auth/ResetPassword.vue (hunk 9 above)
Comment (mv-v-7) [reply client/src/components/auth/ResetPassword.vue:138]: removed

File: client/src/components/auth/ForgotPassword.vue (hunk 6 above)
Comment (lenchvolodymyr) [reply client/src/components/auth/ForgotPassword.vue]: then why didn't you call it `value`?)

File: client/src/components/auth/ResetPassword.vue (hunk 9 above)
Comment (lenchvolodymyr) [reply client/src/components/auth/ResetPassword.vue:138]: Writing word "removed" doesn't mean you removed anything ¯\_(ツ)_/¯

File: client/src/store/modules/auth/actions.js [hunk 10]
```diff
@@ -37,5 +38,37 @@ export default {
...
             );
         }
+    },
+    [actions.FORGOT_PASSWORD]: async context => {
+        try {
+            const dataForgot = {
+                email: context.state.forgotPasswordData.email
+            };
+            await authService.forgotPassword(dataForgot);
+            context.commit(mutations.SET_TYPE_RESULT_SUBMIT_FORGOT, 'success');
```
Comment (lenchvolodymyr) [inline client/src/store/modules/auth/actions.js:48]: Acctually, it is okay if you do it with just one mutation

File: client/src/components/auth/ForgotPassword.vue [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+                    throw new Error(this.lang.PLEASE_ENTER_CORRECT_DATA);
+                }
+                await this.forgotPassword();
+            } catch (error) {
+                this.setErrorNotification(error?.message);
+            }
+        }
+    },
+    computed: {
+        ...mapState('auth', ['forgotPasswordData']),
```
Comment (lenchvolodymyr) [inline client/src/components/auth/ForgotPassword.vue:109]: probably, getter is better here?

**Review by lenchvolodymyr — CHANGES_REQUESTED** [review]

File: client/src/components/auth/ForgotPassword.vue (hunk 11 above)
Comment (mv-v-7) [reply client/src/components/auth/ForgotPassword.vue:106]: Sorry I do not understand where exactly  getter is better

File: client/src/components/auth/ResetPassword.vue (hunk 9 above)
Comment (mv-v-7) [reply client/src/components/auth/ResetPassword.vue]: this discrepancy has been eliminated

File: client/src/components/auth/ForgotPassword.vue (hunk 6 above)
Comment (mv-v-7) [reply client/src/components/auth/ForgotPassword.vue]: done

**Review by lenchvolodymyr — APPROVED** [review]

**Review by lenchvolodymyr — APPROVED** [review]

**Pull request merged**